backend/CSEgen.ml

```diff
@@ -122,6 +133,42 @@ let find_equation op_class n rhs =
   with Not_found ->
     None
 
+(** [valnum_operand] and [valnum_operands] return an array of value numbers
+    associated with the registers referred to by the operands. Similar
+    to [valnum_reg] and [valnum_regs]. *)
+let valnum_operand n operand =
+  let (n1, (rhs_operand : rhs_operand), op_class) =
+    match (operand : Mach.operand) with
+    | Ireg r ->
+      let (n1,v1) = valnum_reg n r in
+      (n1, Vreg v1, Op_pure)
+    | Imem { chunk; addr; reg } ->
+      let (n1, reg) = valnum_regs n reg in
+      (n1, Vmem { chunk; addr; reg }, Op_load)
+    | Iimm i -> (n, Vimm i, Op_pure)
+    | Iimmf f -> (n, Vimmf f, Op_pure)
```

Comment:
Actually, this is fine because the load is from constant memory (never written to). Similarly, `class_of_operation` for `Iconst_float` is `Op_pure`. 
The CR is overly conservative, and I will fix it.